browser/tor/tor_navigation_throttle_unittest.cc

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/tor/tor_navigation_throttle.h"
+
+#include <utility>
+
+#include "brave/common/tor/pref_names.h"
+#include "chrome/browser/prefs/browser_prefs.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/test/base/chrome_render_view_host_test_harness.h"
+#include "chrome/test/base/testing_profile.h"
+#include "components/prefs/pref_service.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "content/public/browser/content_browser_client.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/test/mock_navigation_handle.h"
+#include "content/public/test/web_contents_tester.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+using content::NavigationThrottle;
+
+namespace tor {
+
+class TorNavigationThrottleUnitTest
+    : public ChromeRenderViewHostTestHarness {
+ public:
+  TorNavigationThrottleUnitTest() = default;
+  ~TorNavigationThrottleUnitTest() override = default;
+
+  void SetUp() override {
+    ChromeRenderViewHostTestHarness::SetUp();
+  }
+
+  void TearDown() override {
+    ChromeRenderViewHostTestHarness::TearDown();
+  }
+
+  content::BrowserContext* CreateBrowserContext() override {
+    TestingProfile::Builder builder;
+    auto prefs =
+        std::make_unique<sync_preferences::TestingPrefServiceSyncable>();
+    RegisterUserProfilePrefs(prefs->registry());
+    prefs->registry()->
+      RegisterBooleanPref(prefs::kProfileUsingTor, false);
+    builder.SetPrefService(std::move(prefs));
+    return builder.Build().release();
+  }
+
+  content::RenderFrameHostTester* render_frame_host_tester(
+      content::RenderFrameHost* host) {
+    return content::RenderFrameHostTester::For(host);
+  }
+
+  content::WebContentsTester* web_contents_tester() {
+    return content::WebContentsTester::For(web_contents());
+  }
+ private:
+  DISALLOW_COPY_AND_ASSIGN(TorNavigationThrottleUnitTest);
+};
+
+// Tests TorNavigationThrottle::MaybeCreateThrottleFor with tor enabled/disabled
+TEST_F(TorNavigationThrottleUnitTest, Instantiation) {
+  profile()->GetPrefs()->SetBoolean(prefs::kProfileUsingTor, true);
```

Comment:
don't we have a helper to create a tor profile? This pref will go away with @yrliou's changes

Reply:
ok, I will use `TorUnittestProfileManager` to create one

Reply:
addressed in b639b31aa7f967639983098bfe683110bca787f4